libs/clients/work-machines/src/lib/providers.ts

```diff
@@ -1,14 +1,93 @@
+import {
+  XRoadConfig,
+  IdsClientConfig,
+  LazyDuringDevScope,
+} from '@island.is/nest/config'
+import { ConfigType } from '@nestjs/config'
 import {
   Configuration,
+  MachineCategoryApi,
+  MachineOwnerChangeApi,
+  MachineSupervisorChangeApi,
   MachinesApi,
   MachinesDocumentApi,
 } from '../../gen/fetch'
-import { ApiConfig } from './api.config'
+import { WorkMachinesClientConfig } from './workMachines.config'
+import { createEnhancedFetch } from '@island.is/clients/middlewares'
+
+const ConfigFactory = (
+  xroadConfig: ConfigType<typeof XRoadConfig>,
+  config: ConfigType<typeof WorkMachinesClientConfig>,
+  idsClientConfig: ConfigType<typeof IdsClientConfig>,
+  acceptHeader: string,
+) => ({
+  fetchApi: createEnhancedFetch({
+    name: 'clients-work-machines',
+    organizationSlug: 'vinnueftirlitid',
+    logErrorResponseBody: true,
+    autoAuth: idsClientConfig.isConfigured
+      ? {
+          mode: 'tokenExchange',
+          issuer: idsClientConfig.issuer,
+          clientId: idsClientConfig.clientId,
+          clientSecret: idsClientConfig.clientSecret,
+          scope: config.fetch.scope,
+        }
+      : undefined,
+  }),
+  basePath: `${xroadConfig.xRoadBasePath}/r1/${config.xRoadServicePath}`,
+  headers: {
+    'X-Road-Client': xroadConfig.xRoadClient,
+    Accept: acceptHeader,
+  },
+})
 
-export const apiProviders = [MachinesApi, MachinesDocumentApi].map((Api) => ({
-  provide: Api,
-  useFactory: (configuration: Configuration) => {
-    return new Api(configuration)
+export class CustomMachineApi extends MachinesApi {}
+
+export const apiProviders = [
+  {
+    api: MachinesApi,
+    provide: MachinesApi,
+    acceptHeader: 'application/vnd.ver.machines.hateoas.v1+json',
+  },
+  {
+    api: MachinesApi,
+    provide: CustomMachineApi,
```

Comment:
What is the `CustomMachineApi` providing additionally to the standard `MachinesApi` ?

Reply:
Its looks like after the latest change from vinnueftirlitinu they have changed their Apis. Because they used to have different acceptHeaders from the getting list of machines and single from the same api. I will remove the custom one

Reply:
This is not removed?

Reply:
No, you are right. I removed it by mistake when you commented this and broke this.
But the real reason for this is that there are two endpoints on this MachinesApi that have different acceptHeaders: one is 'application/vnd.ver.machine.hateoas.v1+json' other is 'application/vnd.ver.machines.hateoas.v1+json'. So I needed to add a custom api for the request with the different acceptHeader

Reply:
Ahh I see.